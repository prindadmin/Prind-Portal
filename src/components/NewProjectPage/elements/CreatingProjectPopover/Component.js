import React, { Component } from 'react'
//import PropTypes from 'prop-types'

import PopOverHandler from '../../../common/popOverHandler'

import {
  Spinner,
  Intent,
} from '@blueprintjs/core'

import * as strings from '../../../../data/Strings'

export class Element extends Component {
  static propTypes = {
  }

  getContent = () => {
    return (
      <React.Fragment>
        {strings.PROJECT_CREATION_BILLING_WARNING}
        <Spinner size='150' intent={Intent.DANGER}/>
      </React.Fragment>
    )
  }

  render () {
    return (
      <PopOverHandler>
        <div id='popup-greyer' onClick={(e) => e.stopPropagation()}>
          <div id='project-creation-popover'>
            <div id='popup-box'>
              <div className='project-creation-popover-container' onClick={(e) => e.stopPropagation()}>
                <div className='element-title'>
                  {strings.PROJECT_CREATION_IN_PROGRESS}
                </div>
                <div className='element-description'>
                  {this.getContent()}
                </div>
              </div>
            </div>
          </div>
        </div>
      </PopOverHandler>
    )
  }

}

export default Element
