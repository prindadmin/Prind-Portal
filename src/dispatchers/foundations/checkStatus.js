import axios from 'axios'
import https from 'https'

export default function( values ) {

  return new Promise((resolve, reject) => {
    const instance = axios.create({
      httpsAgent: new https.Agent({
        rejectUnauthorized: false
      }),
      headers: {
        'Content-Type': 'application/json',
      }
    });

    instance.post(`${process.env.REACT_APP_FOUNDATIONS_CHECK_STATUS}`, values)
    .then(res => {
      //console.log("success")
      //console.log(res)
      resolve(res)
    })
    .catch((error) => {
      //console.log("error")
      //console.error(error)
      reject(error)
    })
  })
}
