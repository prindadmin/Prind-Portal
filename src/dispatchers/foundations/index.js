import selfSignFileDispatcher from './selfSignFile'
import rejectSignatureRequestDispatcher from './rejectSignatureRequest'

export {
  selfSignFileDispatcher,
  rejectSignatureRequestDispatcher,
}
