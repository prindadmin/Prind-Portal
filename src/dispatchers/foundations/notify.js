import axios from 'axios'
import https from 'https'

export default function( values ) {

  return new Promise((resolve, reject) => {
    const instance = axios.create({
      httpsAgent: new https.Agent({
        rejectUnauthorized: false
      }),
      headers: {
        'Content-Type': 'application/json',
        'Authorization': '9284a74c-4382-4a25-91be-9b9534464389'
      }
    });

    instance.post(`${process.env.REACT_APP_FOUNDATIONS_SEND_NOTIFICATION}`, values)
    .then(res => {
      //console.log("success")
      //console.log(res)
      resolve(res)
    })
    .catch((error) => {
      //console.log("error")
      //console.error(error)
      reject(error)
    })
  })
}
