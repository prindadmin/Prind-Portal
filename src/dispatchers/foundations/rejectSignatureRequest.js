import { Auth } from 'aws-amplify';
import API from '@aws-amplify/api';

// Fixed values for the API request
const apiName = process.env.REACT_APP_API_NAME

async function RejectSignatureRequest(requestDetails) {

  // Build path for request
  const { projectID, pageName, fieldID } = requestDetails
  const path = `/document/${projectID}/${pageName}/${fieldID}/sign-on-foundations`

  // Get the current session and the identity jwtToken
  const identityToken = await Auth.currentSession()
    .then(credentials => {
        return credentials.idToken.jwtToken
      })

  return new Promise((resolve, reject) => {

    // Create the header for the request
    const myInit = {
        headers: {
          Authorization: identityToken
        },
        body: {
          accepted: false,
        },
        response: false,
    }

    // Send the request
    API.post(apiName, path, myInit)
      .then(response => {
        if (response.Error) {
          reject(response)
          return;
        }
        resolve(response)
      })
      .catch(error => {
        //console.log(error.response);
        reject(error)
     })
   })
}

export default RejectSignatureRequest
