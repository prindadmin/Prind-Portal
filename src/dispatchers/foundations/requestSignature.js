import axios from 'axios'
import https from 'https'

export default function(identityToken, projectID, pageName, fieldID, fieldDetails, members) {

  return new Promise((resolve, reject) => {

    const instance = axios.create({
      httpsAgent: new https.Agent({
        rejectUnauthorized: false
      }),
      headers: {
        'Content-Type': 'application/json',
        'Authorization': identityToken,
      }
    });

    const values = {
      projectID,
      pageName,
      fieldID,
      fieldDetails, 
      members
    }

    console.log(values)

    instance.post(`${process.env.REACT_APP_API_LOCAL_ENDPOINT}/foundations/request-signature.json`, values)
    .then(res => {
      resolve(res)
    })
    .catch((error) => {
      console.log(error)
      reject(error)
    })
  })
}
