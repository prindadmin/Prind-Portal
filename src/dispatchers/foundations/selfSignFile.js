import axios from 'axios'
import https from 'https'

// TODO: Implement API endpoint

export default function(identityToken, fieldDetails) {

  return new Promise((resolve, reject) => {

    const instance = axios.create({
      httpsAgent: new https.Agent({
        rejectUnauthorized: false
      }),
      headers: {
        'Content-Type': 'application/json',
        'Authorization': identityToken,
      }
    });

    instance.get(`${process.env.REACT_APP_API_LOCAL_ENDPOINT}/foundations/self-sign.json`, fieldDetails)
    .then(res => {
      resolve(res)
    })
    .catch((error) => {
      console.error(error)
      reject(error)
    })
  })
}
