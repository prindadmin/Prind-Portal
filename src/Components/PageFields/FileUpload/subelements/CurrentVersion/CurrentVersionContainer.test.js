//import { Suspense } from 'react';
import { Provider, connect } from 'react-redux'
import { shallow, mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import { HashRouter as Router } from "react-router-dom";

import Component from './CurrentVersionContainer'
import * as foundations from '../../../../../Reducers/foundations'

const mockStore = configureStore([]);
var component;
var store;
var props;


beforeAll(() => {

});

beforeEach(() => {
  store = mockStore({
    user: {
      username: "aaaa-aaaaaaaa-aaaa-aaaa-aaaa",
      details: {
        foundationsID: "1234"
      }
    },
    projects: {
      accessibleProjects: {
        projectCreator: [],
        projectRole: []
      },
      chosenProject: {
        projectName: "Test Project 1",
        projectId: "TestProjectID",
        projectType: "CDM2015",
      },
      fetching: false,
      error: null,
    },
  });

  const props = {
    details: {
      uploadName: "File123.txt",
      uploadedDateTime: "2021-04-06T07:06:23",
      uploadedBy: "Testy McTesterson",
      proofLink: "https://prind.tech",
      ver: "0"
    },
    pageName: "design",
    fieldID: "1",
    editable: true,
  };

  component = mount(
    <Provider store={store}>
      <Router>
        <Component {...props} />
      </Router>
    </Provider>
  );
});


it('CurrentVersionContainer - should render', () => {
  expect(component).toMatchSnapshot();
});


it('CurrentVersionContainer - Click self-sign button', () => {
  component.find('#self-sign-button').simulate('click')
  expect(component).toMatchSnapshot();
});
