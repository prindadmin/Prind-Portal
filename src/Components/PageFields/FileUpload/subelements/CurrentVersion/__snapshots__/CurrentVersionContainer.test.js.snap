// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentVersionContainer - Click self-sign button 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <HashRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <withRouter(Connect(CurrentVersion))
        details={
          Object {
            "proofLink": "https://prind.tech",
            "uploadName": "File123.txt",
            "uploadedBy": "Testy McTesterson",
            "uploadedDateTime": "2021-04-06T07:06:23",
            "ver": "0",
          }
        }
        editable={true}
        fieldID="1"
        pageName="design"
      >
        <Connect(CurrentVersion)
          details={
            Object {
              "proofLink": "https://prind.tech",
              "uploadName": "File123.txt",
              "uploadedBy": "Testy McTesterson",
              "uploadedDateTime": "2021-04-06T07:06:23",
              "ver": "0",
            }
          }
          editable={true}
          fieldID="1"
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "createHref": [Function],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
          location={
            Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            }
          }
          match={
            Object {
              "isExact": true,
              "params": Object {},
              "path": "/",
              "url": "/",
            }
          }
          pageName="design"
        >
          <CurrentVersion
            details={
              Object {
                "proofLink": "https://prind.tech",
                "uploadName": "File123.txt",
                "uploadedBy": "Testy McTesterson",
                "uploadedDateTime": "2021-04-06T07:06:23",
                "ver": "0",
              }
            }
            editable={true}
            fieldID="1"
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "createHref": [Function],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            location={
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              }
            }
            match={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
            pageName="design"
            projects={
              Object {
                "accessibleProjects": Object {
                  "projectCreator": Array [],
                  "projectRole": Array [],
                },
                "chosenProject": Object {
                  "projectId": "TestProjectID",
                  "projectName": "Test Project 1",
                  "projectType": "CDM2015",
                },
                "error": null,
                "fetching": false,
              }
            }
            selfSignFile={[Function]}
            user={
              Object {
                "details": Object {
                  "foundationsID": "1234",
                },
                "username": "aaaa-aaaaaaaa-aaaa-aaaa-aaaa",
              }
            }
          >
            <div
              id="current-version-container"
            >
              <div
                className="element-title"
              >
                Current Version
              </div>
              <div
                className="details-table"
              >
                <h4>
                  File Name: 
                </h4>
                <div>
                  File123.txt
                </div>
                <h4>
                  Uploaded At: 
                </h4>
                <div>
                  Invalid Date
                </div>
                <h4>
                  Uploaded By: 
                </h4>
                <div>
                  User without Foundations
                </div>
                <h4>
                  Proof: 
                </h4>
                <div>
                  <div
                    id="proof-link-container"
                    onClick={[Function]}
                  >
                    <a
                      href="undefined/entries/prind.tech"
                      id="proof-link"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      Link to Proof
                    </a>
                  </div>
                </div>
              </div>
              <div
                className="row button-row"
                style={
                  Object {
                    "marginTop": "1rem",
                  }
                }
              >
                <input
                  className="button"
                  disabled={false}
                  id="request-signatures-button"
                  onClick={[Function]}
                  type="submit"
                  value="Request Signatures"
                />
                <input
                  className="button"
                  disabled={false}
                  id="self-sign-button"
                  onClick={[Function]}
                  type="submit"
                  value="Self Sign Document"
                />
              </div>
              <div
                className="download-box-cell"
                onClick={[Function]}
              >
                <withRouter(Connect(DownloadBox))
                  fieldID="1"
                  fileVersionDetails={
                    Object {
                      "proofLink": "https://prind.tech",
                      "uploadName": "File123.txt",
                      "uploadedBy": "Testy McTesterson",
                      "uploadedDateTime": "2021-04-06T07:06:23",
                      "ver": "0",
                    }
                  }
                  onDownloadFailure={[Function]}
                  onDownloadSuccess={[Function]}
                  pageName="design"
                  projectId="TestProjectID"
                >
                  <Connect(DownloadBox)
                    fieldID="1"
                    fileVersionDetails={
                      Object {
                        "proofLink": "https://prind.tech",
                        "uploadName": "File123.txt",
                        "uploadedBy": "Testy McTesterson",
                        "uploadedDateTime": "2021-04-06T07:06:23",
                        "ver": "0",
                      }
                    }
                    history={
                      Object {
                        "action": "POP",
                        "block": [Function],
                        "createHref": [Function],
                        "go": [Function],
                        "goBack": [Function],
                        "goForward": [Function],
                        "length": 1,
                        "listen": [Function],
                        "location": Object {
                          "hash": "",
                          "pathname": "/",
                          "search": "",
                          "state": undefined,
                        },
                        "push": [Function],
                        "replace": [Function],
                      }
                    }
                    location={
                      Object {
                        "hash": "",
                        "pathname": "/",
                        "search": "",
                        "state": undefined,
                      }
                    }
                    match={
                      Object {
                        "isExact": true,
                        "params": Object {},
                        "path": "/",
                        "url": "/",
                      }
                    }
                    onDownloadFailure={[Function]}
                    onDownloadSuccess={[Function]}
                    pageName="design"
                    projectId="TestProjectID"
                  >
                    <DownloadBox
                      downloadFile={[Function]}
                      fieldID="1"
                      fileVersionDetails={
                        Object {
                          "proofLink": "https://prind.tech",
                          "uploadName": "File123.txt",
                          "uploadedBy": "Testy McTesterson",
                          "uploadedDateTime": "2021-04-06T07:06:23",
                          "ver": "0",
                        }
                      }
                      history={
                        Object {
                          "action": "POP",
                          "block": [Function],
                          "createHref": [Function],
                          "go": [Function],
                          "goBack": [Function],
                          "goForward": [Function],
                          "length": 1,
                          "listen": [Function],
                          "location": Object {
                            "hash": "",
                            "pathname": "/",
                            "search": "",
                            "state": undefined,
                          },
                          "push": [Function],
                          "replace": [Function],
                        }
                      }
                      location={
                        Object {
                          "hash": "",
                          "pathname": "/",
                          "search": "",
                          "state": undefined,
                        }
                      }
                      match={
                        Object {
                          "isExact": true,
                          "params": Object {},
                          "path": "/",
                          "url": "/",
                        }
                      }
                      onDownloadFailure={[Function]}
                      onDownloadSuccess={[Function]}
                      pageName="design"
                      projectId="TestProjectID"
                      resetDownloadURL={[Function]}
                    >
                      <div
                        className="downloadBox"
                        id="download-box"
                        onClick={[Function]}
                        style={Object {}}
                      >
                        <div>
                          <ItemIcon
                            size="4x"
                            type="download"
                          >
                            <Blueprint3.Icon
                              icon={
                                <FontAwesomeIcon
                                  border={false}
                                  className=""
                                  fixedWidth={false}
                                  flip={null}
                                  icon={
                                    Object {
                                      "icon": Array [
                                        512,
                                        512,
                                        Array [],
                                        "f019",
                                        "M216 0h80c13.3 0 24 10.7 24 24v168h87.7c17.8 0 26.7 21.5 14.1 34.1L269.7 378.3c-7.5 7.5-19.8 7.5-27.3 0L90.1 226.1c-12.6-12.6-3.7-34.1 14.1-34.1H192V24c0-13.3 10.7-24 24-24zm296 376v112c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V376c0-13.3 10.7-24 24-24h146.7l49 49c20.1 20.1 52.5 20.1 72.6 0l49-49H488c13.3 0 24 10.7 24 24zm-124 88c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20zm64 0c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20z",
                                      ],
                                      "iconName": "download",
                                      "prefix": "fas",
                                    }
                                  }
                                  inverse={false}
                                  listItem={false}
                                  mask={null}
                                  pull={null}
                                  pulse={false}
                                  rotation={null}
                                  size="4x"
                                  spin={false}
                                  swapOpacity={false}
                                  symbol={false}
                                  title=""
                                  transform={null}
                                />
                              }
                            >
                              <FontAwesomeIcon
                                border={false}
                                className=""
                                fixedWidth={false}
                                flip={null}
                                icon={
                                  Object {
                                    "icon": Array [
                                      512,
                                      512,
                                      Array [],
                                      "f019",
                                      "M216 0h80c13.3 0 24 10.7 24 24v168h87.7c17.8 0 26.7 21.5 14.1 34.1L269.7 378.3c-7.5 7.5-19.8 7.5-27.3 0L90.1 226.1c-12.6-12.6-3.7-34.1 14.1-34.1H192V24c0-13.3 10.7-24 24-24zm296 376v112c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V376c0-13.3 10.7-24 24-24h146.7l49 49c20.1 20.1 52.5 20.1 72.6 0l49-49H488c13.3 0 24 10.7 24 24zm-124 88c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20zm64 0c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20z",
                                    ],
                                    "iconName": "download",
                                    "prefix": "fas",
                                  }
                                }
                                inverse={false}
                                listItem={false}
                                mask={null}
                                pull={null}
                                pulse={false}
                                rotation={null}
                                size="4x"
                                spin={false}
                                swapOpacity={false}
                                symbol={false}
                                title=""
                                transform={null}
                              >
                                <svg
                                  aria-hidden="true"
                                  className="svg-inline--fa fa-download fa-w-16 fa-4x "
                                  data-icon="download"
                                  data-prefix="fas"
                                  focusable="false"
                                  role="img"
                                  style={Object {}}
                                  viewBox="0 0 512 512"
                                  xmlns="http://www.w3.org/2000/svg"
                                >
                                  <path
                                    d="M216 0h80c13.3 0 24 10.7 24 24v168h87.7c17.8 0 26.7 21.5 14.1 34.1L269.7 378.3c-7.5 7.5-19.8 7.5-27.3 0L90.1 226.1c-12.6-12.6-3.7-34.1 14.1-34.1H192V24c0-13.3 10.7-24 24-24zm296 376v112c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V376c0-13.3 10.7-24 24-24h146.7l49 49c20.1 20.1 52.5 20.1 72.6 0l49-49H488c13.3 0 24 10.7 24 24zm-124 88c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20zm64 0c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20z"
                                    fill="currentColor"
                                    style={Object {}}
                                  />
                                </svg>
                              </FontAwesomeIcon>
                            </Blueprint3.Icon>
                          </ItemIcon>
                          <p>
                            Download
                          </p>
                        </div>
                      </div>
                    </DownloadBox>
                  </Connect(DownloadBox)>
                </withRouter(Connect(DownloadBox))>
              </div>
            </div>
          </CurrentVersion>
        </Connect(CurrentVersion)>
      </withRouter(Connect(CurrentVersion))>
    </Router>
  </HashRouter>
</Provider>
`;

exports[`CurrentVersionContainer - should render 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <HashRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <withRouter(Connect(CurrentVersion))
        details={
          Object {
            "proofLink": "https://prind.tech",
            "uploadName": "File123.txt",
            "uploadedBy": "Testy McTesterson",
            "uploadedDateTime": "2021-04-06T07:06:23",
            "ver": "0",
          }
        }
        editable={true}
        fieldID="1"
        pageName="design"
      >
        <Connect(CurrentVersion)
          details={
            Object {
              "proofLink": "https://prind.tech",
              "uploadName": "File123.txt",
              "uploadedBy": "Testy McTesterson",
              "uploadedDateTime": "2021-04-06T07:06:23",
              "ver": "0",
            }
          }
          editable={true}
          fieldID="1"
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "createHref": [Function],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
          location={
            Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            }
          }
          match={
            Object {
              "isExact": true,
              "params": Object {},
              "path": "/",
              "url": "/",
            }
          }
          pageName="design"
        >
          <CurrentVersion
            details={
              Object {
                "proofLink": "https://prind.tech",
                "uploadName": "File123.txt",
                "uploadedBy": "Testy McTesterson",
                "uploadedDateTime": "2021-04-06T07:06:23",
                "ver": "0",
              }
            }
            editable={true}
            fieldID="1"
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "createHref": [Function],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            location={
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              }
            }
            match={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
            pageName="design"
            projects={
              Object {
                "accessibleProjects": Object {
                  "projectCreator": Array [],
                  "projectRole": Array [],
                },
                "chosenProject": Object {
                  "projectId": "TestProjectID",
                  "projectName": "Test Project 1",
                  "projectType": "CDM2015",
                },
                "error": null,
                "fetching": false,
              }
            }
            selfSignFile={[Function]}
            user={
              Object {
                "details": Object {
                  "foundationsID": "1234",
                },
                "username": "aaaa-aaaaaaaa-aaaa-aaaa-aaaa",
              }
            }
          >
            <div
              id="current-version-container"
            >
              <div
                className="element-title"
              >
                Current Version
              </div>
              <div
                className="details-table"
              >
                <h4>
                  File Name: 
                </h4>
                <div>
                  File123.txt
                </div>
                <h4>
                  Uploaded At: 
                </h4>
                <div>
                  Invalid Date
                </div>
                <h4>
                  Uploaded By: 
                </h4>
                <div>
                  User without Foundations
                </div>
                <h4>
                  Proof: 
                </h4>
                <div>
                  <div
                    id="proof-link-container"
                    onClick={[Function]}
                  >
                    <a
                      href="undefined/entries/prind.tech"
                      id="proof-link"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      Link to Proof
                    </a>
                  </div>
                </div>
              </div>
              <div
                className="row button-row"
                style={
                  Object {
                    "marginTop": "1rem",
                  }
                }
              >
                <input
                  className="button"
                  disabled={false}
                  id="request-signatures-button"
                  onClick={[Function]}
                  type="submit"
                  value="Request Signatures"
                />
                <input
                  className="button"
                  disabled={false}
                  id="self-sign-button"
                  onClick={[Function]}
                  type="submit"
                  value="Self Sign Document"
                />
              </div>
              <div
                className="download-box-cell"
                onClick={[Function]}
              >
                <withRouter(Connect(DownloadBox))
                  fieldID="1"
                  fileVersionDetails={
                    Object {
                      "proofLink": "https://prind.tech",
                      "uploadName": "File123.txt",
                      "uploadedBy": "Testy McTesterson",
                      "uploadedDateTime": "2021-04-06T07:06:23",
                      "ver": "0",
                    }
                  }
                  onDownloadFailure={[Function]}
                  onDownloadSuccess={[Function]}
                  pageName="design"
                  projectId="TestProjectID"
                >
                  <Connect(DownloadBox)
                    fieldID="1"
                    fileVersionDetails={
                      Object {
                        "proofLink": "https://prind.tech",
                        "uploadName": "File123.txt",
                        "uploadedBy": "Testy McTesterson",
                        "uploadedDateTime": "2021-04-06T07:06:23",
                        "ver": "0",
                      }
                    }
                    history={
                      Object {
                        "action": "POP",
                        "block": [Function],
                        "createHref": [Function],
                        "go": [Function],
                        "goBack": [Function],
                        "goForward": [Function],
                        "length": 1,
                        "listen": [Function],
                        "location": Object {
                          "hash": "",
                          "pathname": "/",
                          "search": "",
                          "state": undefined,
                        },
                        "push": [Function],
                        "replace": [Function],
                      }
                    }
                    location={
                      Object {
                        "hash": "",
                        "pathname": "/",
                        "search": "",
                        "state": undefined,
                      }
                    }
                    match={
                      Object {
                        "isExact": true,
                        "params": Object {},
                        "path": "/",
                        "url": "/",
                      }
                    }
                    onDownloadFailure={[Function]}
                    onDownloadSuccess={[Function]}
                    pageName="design"
                    projectId="TestProjectID"
                  >
                    <DownloadBox
                      downloadFile={[Function]}
                      fieldID="1"
                      fileVersionDetails={
                        Object {
                          "proofLink": "https://prind.tech",
                          "uploadName": "File123.txt",
                          "uploadedBy": "Testy McTesterson",
                          "uploadedDateTime": "2021-04-06T07:06:23",
                          "ver": "0",
                        }
                      }
                      history={
                        Object {
                          "action": "POP",
                          "block": [Function],
                          "createHref": [Function],
                          "go": [Function],
                          "goBack": [Function],
                          "goForward": [Function],
                          "length": 1,
                          "listen": [Function],
                          "location": Object {
                            "hash": "",
                            "pathname": "/",
                            "search": "",
                            "state": undefined,
                          },
                          "push": [Function],
                          "replace": [Function],
                        }
                      }
                      location={
                        Object {
                          "hash": "",
                          "pathname": "/",
                          "search": "",
                          "state": undefined,
                        }
                      }
                      match={
                        Object {
                          "isExact": true,
                          "params": Object {},
                          "path": "/",
                          "url": "/",
                        }
                      }
                      onDownloadFailure={[Function]}
                      onDownloadSuccess={[Function]}
                      pageName="design"
                      projectId="TestProjectID"
                      resetDownloadURL={[Function]}
                    >
                      <div
                        className="downloadBox"
                        id="download-box"
                        onClick={[Function]}
                        style={Object {}}
                      >
                        <div>
                          <ItemIcon
                            size="4x"
                            type="download"
                          >
                            <Blueprint3.Icon
                              icon={
                                <FontAwesomeIcon
                                  border={false}
                                  className=""
                                  fixedWidth={false}
                                  flip={null}
                                  icon={
                                    Object {
                                      "icon": Array [
                                        512,
                                        512,
                                        Array [],
                                        "f019",
                                        "M216 0h80c13.3 0 24 10.7 24 24v168h87.7c17.8 0 26.7 21.5 14.1 34.1L269.7 378.3c-7.5 7.5-19.8 7.5-27.3 0L90.1 226.1c-12.6-12.6-3.7-34.1 14.1-34.1H192V24c0-13.3 10.7-24 24-24zm296 376v112c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V376c0-13.3 10.7-24 24-24h146.7l49 49c20.1 20.1 52.5 20.1 72.6 0l49-49H488c13.3 0 24 10.7 24 24zm-124 88c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20zm64 0c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20z",
                                      ],
                                      "iconName": "download",
                                      "prefix": "fas",
                                    }
                                  }
                                  inverse={false}
                                  listItem={false}
                                  mask={null}
                                  pull={null}
                                  pulse={false}
                                  rotation={null}
                                  size="4x"
                                  spin={false}
                                  swapOpacity={false}
                                  symbol={false}
                                  title=""
                                  transform={null}
                                />
                              }
                            >
                              <FontAwesomeIcon
                                border={false}
                                className=""
                                fixedWidth={false}
                                flip={null}
                                icon={
                                  Object {
                                    "icon": Array [
                                      512,
                                      512,
                                      Array [],
                                      "f019",
                                      "M216 0h80c13.3 0 24 10.7 24 24v168h87.7c17.8 0 26.7 21.5 14.1 34.1L269.7 378.3c-7.5 7.5-19.8 7.5-27.3 0L90.1 226.1c-12.6-12.6-3.7-34.1 14.1-34.1H192V24c0-13.3 10.7-24 24-24zm296 376v112c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V376c0-13.3 10.7-24 24-24h146.7l49 49c20.1 20.1 52.5 20.1 72.6 0l49-49H488c13.3 0 24 10.7 24 24zm-124 88c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20zm64 0c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20z",
                                    ],
                                    "iconName": "download",
                                    "prefix": "fas",
                                  }
                                }
                                inverse={false}
                                listItem={false}
                                mask={null}
                                pull={null}
                                pulse={false}
                                rotation={null}
                                size="4x"
                                spin={false}
                                swapOpacity={false}
                                symbol={false}
                                title=""
                                transform={null}
                              >
                                <svg
                                  aria-hidden="true"
                                  className="svg-inline--fa fa-download fa-w-16 fa-4x "
                                  data-icon="download"
                                  data-prefix="fas"
                                  focusable="false"
                                  role="img"
                                  style={Object {}}
                                  viewBox="0 0 512 512"
                                  xmlns="http://www.w3.org/2000/svg"
                                >
                                  <path
                                    d="M216 0h80c13.3 0 24 10.7 24 24v168h87.7c17.8 0 26.7 21.5 14.1 34.1L269.7 378.3c-7.5 7.5-19.8 7.5-27.3 0L90.1 226.1c-12.6-12.6-3.7-34.1 14.1-34.1H192V24c0-13.3 10.7-24 24-24zm296 376v112c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V376c0-13.3 10.7-24 24-24h146.7l49 49c20.1 20.1 52.5 20.1 72.6 0l49-49H488c13.3 0 24 10.7 24 24zm-124 88c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20zm64 0c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20z"
                                    fill="currentColor"
                                    style={Object {}}
                                  />
                                </svg>
                              </FontAwesomeIcon>
                            </Blueprint3.Icon>
                          </ItemIcon>
                          <p>
                            Download
                          </p>
                        </div>
                      </div>
                    </DownloadBox>
                  </Connect(DownloadBox)>
                </withRouter(Connect(DownloadBox))>
              </div>
            </div>
          </CurrentVersion>
        </Connect(CurrentVersion)>
      </withRouter(Connect(CurrentVersion))>
    </Router>
  </HashRouter>
</Provider>
`;
